src/overlays/actors/ovl_Bg_Spot11_Oasis/z_bg_spot11_oasis.c

```diff
@@ -19,25 +25,136 @@ const ActorInit Bg_Spot11_Oasis_InitVars = {
+s16 D_808B2E10[][2] = {
+    { 1260, 2040 }, { 1259, 1947 }, { 1135, 1860 }, { 1087, 1912 }, { 1173, 2044 },
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F(scale, 1, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 3000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 1200, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+Vec3f D_808B2E34[] = {
+    { 0.0f, -100.0f, 0.0f },   { 100.0f, -80.0f, -50.0f }, { -50.0f, -80.0f, -100.0f },
+    { -75.0f, -90.0f, 90.0f }, { 30.0f, -100.0f, 40.0f },
+};
+
+extern Gfx D_06000870[];
+
+void func_808B27F0(GlobalContext* globalCtx, s16 arg1) {
+    WaterBox* waterBoxes;
+    waterBoxes = globalCtx->colCtx.stat.colHeader->waterBoxes;
+    waterBoxes->unk_02 = arg1;
+}
+
+s32 func_808B280C(GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    Vec3f sp58;
+    Vec3f sp4C;
+    Vec3f sp40;
+    s32 i;
+
+    sp58.x = D_808B2E10[0][0];
+    sp58.z = D_808B2E10[0][1];
+    sp58.y = 0.0f;
+
+    sp4C.y = 0.0f;
+    sp40.y = 0.0f;
+
+    for (i = 1; i < ARRAY_COUNT(D_808B2E10) - 1; i++) {
+        sp4C.x = D_808B2E10[i][0];
+        sp4C.z = D_808B2E10[i][1];
+        sp40.x = D_808B2E10[i + 1][0];
+        sp40.z = D_808B2E10[i + 1][1];
+        if (Math3D_TriChkPointParaYSlopedY(&sp58, &sp4C, &sp40, player->actor.posRot.pos.z,
+                                           player->actor.posRot.pos.x)) {
+            return 1;
+        }
+    }
+    return 0;
+}
+
+void BgSpot11Oasis_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot11Oasis* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    func_808B2970(this);
+    this->actor.posRot.pos.y = -100.0f;
+    func_808B27F0(globalCtx, -100);
+}
+
+void func_808B2970(BgSpot11Oasis* this) {
+    this->actionFunc = func_808B2980;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot11_Oasis/BgSpot11Oasis_Init.s")
+void func_808B2980(BgSpot11Oasis* this, GlobalContext* globalCtx) {
+    if (Flags_GetEnv(globalCtx, 5) && func_808B280C(globalCtx)) {
+        func_800800F8(globalCtx, 0x1036, -99, &this->actor, 0);
+        func_808B29E0(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot11_Oasis/func_808B2970.s")
+void func_808B29E0(BgSpot11Oasis* this) {
+    this->actionFunc = func_808B29F0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot11_Oasis/func_808B2980.s")
+void func_808B29F0(BgSpot11Oasis* this, GlobalContext* globalCtx) {
+    if (Math_ApproxF(&this->actor.posRot.pos.y, 0.0f, 0.7f)) {
+        func_808B2AA8(this);
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ELF, this->actor.posRot.pos.x,
+                    this->actor.posRot.pos.y + 40.0f, this->actor.posRot.pos.z, 0, 0, 0, 4);
+        func_80078884(NA_SE_SY_CORRECT_CHIME);
+    }
+    func_808B27F0(globalCtx, (s16)this->actor.posRot.pos.y);
```

Comment:
Is the cast needed here?
```suggestion
    func_808B27F0(globalCtx, this->actor.posRot.pos.y);
```